Safewaan/HonkedIn
# Gabbytang/create events

## Trello ticket link

[As a user, I want to be able to create events that other users can attend.](https://trello.com/c/9qzkpF4s/2-as-a-user-i-want-to-be-able-to-create-events-that-other-users-can-attend )



## Implementation description
* Users can now create events that are stored and saved in the database 
* Current event fields are as follows: Name, Description, Location, Date, Creator
* Creator field is automatically populated based on the user that is logged in and creating the given event
* Users should get errors if they try to submit an incomplete form in event creation.
* Users can go to the Events page to see all events that have dates that have not already passed
* CURRENTLY NOT IMPLEMENTED: Event participant tracking on both the creator and participant side has not yet been implemented. 


## Steps to test
1. Verify that you can go to the /create-event page by adding that extension to your localhost:3000 URL 
2. Verify that you can enter values in the fields. 
3. If you leave fields blank, you should be getting errors and not see the form reset (meaning that the submission has not passed) 
4. Once the form is complete, clicking SUBMIT should reset the form and create an Event record. 
5. Check that the event you have just created is now in the _Events_ table in the db by running the query 
```
use shchowdh; 
select firstName, lastName, events.name, events.description, events.location, events.date
from users, events
where events.creatorID = users.id
and date>=curdate()
order by date 
```
7. Back at the front end, go to the Events page by switching the url extension to /events
8. You should be able to see a list of all events with dates that have not passed yet
9. Your created event should be there with all the field values you input during creation. 



## What should reviewers focus on?
* Make sure that if you try to leave fields blank, you will get an error if you try to submit it and nothing should go to the db 
* The error flags are slightly bugged, it is expected that when you open the /create-events page you will see an error flagging the empty Name field.
